common/hashing.cpp

```diff
@@ -0,0 +1,77 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "common/hashing.h"
+
+namespace Carbon {
+
+auto Hasher::HashSizedBytesLarge(Hasher hash, llvm::ArrayRef<std::byte> bytes)
+    -> Hasher {
+  const std::byte* data_ptr = bytes.data();
+  const ssize_t size = bytes.size();
+  CARBON_DCHECK(size > 32);
+
+  __builtin_prefetch(data_ptr, 0, 0);
```

Comment:
I'm trying not to comment much, but I do want to remark here, this kind of comment wouldn't be possible in Carbon: can the comments be adjusted to match Carbon's style of only using line comments? @chandlerc While you've suggested in the past that maybe Carbon would allow more flexible argument name labeling, I do think that "discard after next use" is going a little past that.

https://github.com/carbon-language/carbon-lang/blob/trunk/docs/project/cpp_style_guide.md#syntax-and-formatting